app/views/articles/edit.html.erb

```diff
@@ -5,5 +5,7 @@
   <%= form.text_field :title, value: @article.title %>
   <%= form.label :content %>
   <%= form.text_area :body, id: 'article_content', value: @article.body %>
+  <%= form.label :categories %>
+  <%= form.select(:categories, Category.all.collect {|c| [ c.name, c.id ] }, {}, { multiple: true, id: 'Categories' })%>
```

Comment:
Again, the form fields contain a lot of repetition. Is a partial a solution? `form_with` will know what action you perform, so there's no need to add `value:` to the submit form. And if you have `@article` defined on the `:new` and `:edit` actions, you should be able to use a partial.

Reply:
Also, I'm not a great fan of single letter variable names.

Reply:
I would also set an `@categories = Category.all` instance variable in your controller to avoid calling the db from your view. (Not a requirement though)

Reply:
IMO is easier to read this then to create a partial for just two line of code!

Reply:
Instance variable makes since.